rest/permissions.py

```diff
@@ -44,3 +49,60 @@ def has_permission(self, request, view):
         except PermissionDenied:
             return False
         return True
+
+
+class ViewVocabularyPermission(BasePermission):
+    """Checks view_repo permission on repository and that
+    repository owns vocabulary"""
+
+    def has_permission(self, request, view):
+        try:
+            get_vocabulary(
+                view.kwargs['repo_slug'],
+                request.user.id,
+                view.kwargs['vocab_slug']
+            )
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+        return True
+
+
+class ViewTermPermission(BasePermission):
+    """Checks view_repo permission on repository and that repository owns
+    vocabulary and that vocabulary owns term"""
+
+    def has_permission(self, request, view):
+        try:
+            get_term(
+                view.kwargs['repo_slug'],
+                request.user.id,
+                view.kwargs['vocab_slug'],
+                view.kwargs['term_slug']
+            )
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+        return True
+
+
+class ManageTaxonomyPermission(BasePermission):
+    """Checks manage_taxonomy permission"""
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
```

Comment:
I still don't think this should return `True` for any user. I know that this permission is only ever checked with other permissions, but the [documentation](http://www.django-rest-framework.org/api-guide/permissions/#custom-permissions) seems to recommend doing that check and then still checking for permissions based on whether it's a "destructive" method.

There _is_ one section in [the examples](http://www.django-rest-framework.org/api-guide/permissions/#examples) which shows a blind `return True` for `SAFE_METHODS`, but that example is clearly meant for objects the world can view and only owners can edit, like blog posts, not things like taxonomies which are strictly tied to the repository ownership of the user.